### PR TITLE
Register

### PR DESCRIPTION
**Pull request by KyeBuff**
Custom endpoint created to handle the registration of users. I think the main thing to check the robustness of the process. 

I also did a bit of a tidy up as we had some warnings in the console.

This PR should be reviewed inline with the API register PR.

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -2,22 +2,24 @@ import React from 'react';
 import { Link } from "react-router-dom";
 import ModulesNav from "./ModulesNav";
 
-export default ({user, loggedIn, logOut, userRole}) => (
+export default ({user, loggedIn, logOut, userRole, setRegistering, registering, cancel}) => (
```
Comment (petethomasmusique) [inline src/components/Header.js]: A good convention to get into for a variable that represents a boolean is using the prefix 'is' as in 'isRegistering'. Makes it more readable for other devs ( if(isRegistering) {...} else {...} ), and you know what type of data to expect it to be...

Comment (KyeBuff) [reply src/components/Header.js]: Yeah the flash of instructor view is due to being logged in as an instructor before and the time it takes for the API to return the user role and thus render the correct view. Shouldn't be an issue with students who will never access the instructor view.

Comment (KyeBuff) [reply src/components/Header.js]: Have updated to isRegistering.

**Comment by petethomasmusique** [comment]
So this still logs in newly registered students as an instructor. If I log in with a student created in the backend I don't have the same problem.

**Pull request merged**